x/auth/tx/service.go

```diff
@@ -37,6 +41,79 @@ func NewTxServer(clientCtx client.Context, simulate baseAppSimulateFn, interface
 
 var _ txtypes.ServiceServer = txServer{}
 
+const (
+	eventFormat = "{eventType}.{eventAttribute}={value}"
+)
+
+// TxsByEvents implements the ServiceServer.TxsByEvents RPC method.
+func (s txServer) TxsByEvents(ctx context.Context, req *txtypes.GetTxsEventRequest) (*txtypes.TxsByEventsResponse, error) {
+
+	offset := int(req.Pagination.Offset)
+	limit := int(req.Pagination.Limit)
```

Comment:
Are we certain to not have a limit of zero?

Reply:
I think we should use `DefaultLimit` if the limit provided is `0`.

Reply:
Great! But best to be safe than sorry :)